unison-src/transcripts/idempotent/fix2238.md

```diff
@@ -1,31 +1,42 @@
+``` ucm :hide
+scratch/main> builtins.mergeio
+```
+
 This should not typecheck - the inline `@eval` expression uses abilities.
 
-``` unison
+``` unison :error
 structural ability Abort where abort : x
 
 ex = {{ @eval{abort} }}
 ```
 
-``` ucm
+``` ucm :added-by-ucm
 
   Loading changes detected in scratch.u.
 
   The expression in red needs the {Abort} ability, but this location does not have access to any abilities.
-  
-      3 | ex = {{ @eval{abort} }}
-  
 
+      3 | ex = {{ @eval{abort} }}
 ```
+
 This file should also not typecheck - it has a triple backticks block that uses abilities.
 
-``` ucm
-scratch/main> load unison-src/transcripts/fix2238.u
```

Comment:
Good call, I'm not sure why this was in a separate file to begin with.

Reply:
It’s because transcripts used to only support exactly three `` ` `` for blocks, and that unison code contained triple-`` ` `` , so couldn’t be nested. That was fixed in #5204, but I think this PR needed some changes so that the output was produced with the correct number of `` ` `` (which is necessary for re-running it[^1]).

[^1]: There were a couple cases where the generated output wasn’t valid (like having the wrong number of `` ` `` for a block). Previously, this was easy to overlook, but would only be annoying when viewing rendered output. But now it would break idempotency, so needed to be fixed.